src/components/Contacts/ContactFlow/ContactFlow.tsx

```diff
@@ -122,14 +122,69 @@ export const ContactFlow: React.FC<Props> = ({
     enqueueSnackbar(t('Contact status info updated!'), {
       variant: 'success',
     });
-    if (status.id && taskStatuses[status.id]) {
-      openTaskModal({
-        view: TaskModalEnum.Add,
-        defaultValues: {
-          activityType: taskStatuses[status.id],
-          contactIds: [id],
-        },
-      });
+
+    switch (contactPhase) {
+      case PhaseEnum.Initiation:
+      case PhaseEnum.Appointment:
+      case PhaseEnum.FollowUp:
+      case PhaseEnum.PartnerCare:
+        await hasActiveTask({
+          variables: {
+            accountListId,
+            sortBy: TaskSortEnum.StartAtAsc,
+            first: 1,
+            tasksFilter: {
+              completed: false,
+              contactIds: [id],
+              activityType: contactPhase
+                ? getActivitiesByPhaseType(contactPhase)
+                : [],
+            },
+          },
+          onCompleted(data) {
```

Comment:
I'd get the data from the return of `hasActiveTask` instead of inside `onCompleted` to reduce nesting.